### ISSUE TITLE
Tiled Gallery / Carousel: Comments not displaying under image in carousel

### ISSUE THREAD

**Issue by tellisbethel** [Automattic/jetpack#23728]
### Quick summary

Users are unable to see any comments added to images in their Tiled Gallery carousels. 

I've encountered this on the user's Twenty Eleven (Simple) theme, and on my test sites Blockbase (Atomic, Simple) and Hever (Simple). 

(I'm not aware of any workarounds so I'm selecting the high priority label in accordance with the priority graph. Please feel free to update if needed.)

### Steps to reproduce

1. Add a Tiled Gallery to a post and save
2. Visit the post and select a Tiled Gallery image to open the carousel
3. Select the comments icon to toggle the comments section and add a comment
4. Post comment (approve comment from dashboard if necessary)

### What you expected to happen

I expected the comment to show up as seen in this image taken from the [Jetpack Carousel guide page](https://jetpack.com/support/carousel/#carousel-options): 

![image](https://user-images.githubusercontent.com/1767799/159176378-72c8faaf-0e6a-45b5-9239-68a84fcdb21f.png)

### What actually happened

No comments show. Comments are available for viewing by accessing the Comments page in the dashboard, but not from within the Tiled Gallery Carousel.

### Context

I encountered this issue here: 4861753-zd-woothemes

### Simple, Atomic or both?

Simple, Atomic

### Reproducibility

Consistent

**Comment by jeherve** [comment]
This is caused by #22770 , and impacts both self-hosted sites running the development version of Jetpack, and WordPress.com simple sites since this PR was also deployed in D74923-code.

https://github.com/Automattic/jetpack/blob/e49848ad05f857695ca2e727db0e63c09cec753f/projects/plugins/jetpack/modules/carousel/jetpack-carousel.php#L936-L938

#### Steps to reproduce

1. Go to Posts > Add New
2. Add a gallery, and publish your post.
3. Go to Media > Upload in wp-admin, pick one of the images in the gallery, you just added, click on it, and then click on the link to access the attachment page.
4. On the attachment page, leave a comment under the image.
5. Go to Jetpack > Settings
6. Enable the Carousel feature
7. Access the post you just published.
8. Click on the gallery to open the Carousel modal.
9. In the network tab, look for an Ajax request fetching comments, something like `https://yoursite.com/wp-admin/admin-ajax.php?action=get_attachment_comments`
10. You'll see the response is always `Nonce verification failed`.

cc @coder-karen

**Comment by kraftbj** [comment]
My hunch is this is what's catching https://github.com/Automattic/jetpack/pull/22770/files#diff-20a0b22f28fd1f2b380b5df535f70d1bd90cca42606e3ff5c7d57fba2ffcbfc2R937
